src/umleditor/mvc_model/custom_exceptions.py

```diff
@@ -76,6 +76,32 @@ class RelationDoesNotExistError(Error):
         def __init__(self, source, destination):
             super().__init__(f"Relation between '{source} -> {destination}' does not exist.")
 
+    class InvalidRelationTypeError(Error):
+        """
+        Exception raised when the relation being added has no types.
+        
+        Args:
+            source (Entity): The source of the relation that was being added.
+            destination (Entity): The destination of the relation that was
+                being added.
+
+        """
+        def __init__(self, source, destination):
+            super().__init__(f"Relation between '{source} -> {destination}' has invalid type.")
+
+    class InvalidRelationNewTypeError(Error):
+        """
+        Exception raised when the relation being added has invalid new type.
+        
+        Args:
+            source (Entity): The source of the relation that was being added.
+            destination (Entity): The destination of the relation that was
+                being added.
+
+        """
+        def __init__(self, source, destination):
+            super().__init__(f"Relation between '{source} -> {destination}' has invalid new type.")
```

Comment:
I feel like passing in relation and saying "x is not a valid relation type" would make more sense.

Reply:
I'm operating on the assumption that this error is raised when a relation type is invalid, not when the source or destination are invalid.